src/java/io/bazel/rulesscala/worker/Worker.java

```diff
@@ -11,22 +11,25 @@
 import java.nio.file.Paths;
 import java.security.Permission;
 import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * A base for JVM workers.
  *
- * <p>This supports regular workers as well as persisent workers. It does not (yet) support
- * multiplexed workers.
+ * <p>This supports regular workers as well as persisent workers.
  *
  * <p>Worker implementations should implement the `Worker.Interface` interface and provide a main
  * method that calls `Worker.workerMain`.
  */
 public final class Worker {
 
+  private static ExecutorService executorService = Executors.newCachedThreadPool();
+
   public static interface Interface {
-    public void work(String[] args) throws Exception;
+    public void work(String[] args, PrintStream out, PrintStream err) throws Exception;
```

Comment:
@wix-andriusb why worker interface takes two `PrintStream`s? Implementations are not allowed to touch `stdout` so I think the choice should not be given. And you are passing `stderr` twice anyway. I think minimal worker interface should be `(args, stderr) -> exit_code`

Reply:
It needs a dedicated PrintStream for stdout, otherwise if a worker would do multiple job requests in parallel the std output would interleave.

Are you suggesting to use one PrintStream for both stdout and stderr?

Reply:
Yes I suggest to use single output stream since you are passing same output stream for both parameters anyway https://github.com/bazelbuild/rules_scala/pull/1293/files#diff-ffcbe22d947b0629a710679137348a0a1c7a6b32e9b8cf284803c549ae194b97R99 
Unless I'm completely missing something.

Reply:
fixed this